webui/src/lib/api/index.js

```diff
@@ -118,6 +118,36 @@ export class MergeError extends Error {
 // actual actions:
 class Auth {
 
+    async updatePasswordByToken(token, newPassword) {
+        const response = await fetch(`${API_ENDPOINT}/auth/password`, {
+            headers: new Headers(defaultAPIHeaders),
+            method: 'POST',
+            body: json({token: token, newPassword: newPassword})
+        });
+
+        if (response.status === 401) {
+            throw new AuthorizationError('user unauthorized');
+        }
+        if (response.status !== 201) {
+            throw new Error('unknown authentication error');
```

Comment:
Failed to update password? or something that explain that we didn't managed to update the password.